huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go

```diff
@@ -0,0 +1,293 @@
+				KeepResponseBody: true,
+				OkCodes: []int{
+					200,
+				},
+				MoreHeaders: map[string]string{"Content-Type": "application/json"},
+			}
+
+			applyPolicyResp, err := applyPolicyClient.Request("GET",
+				applyPolicyPath, &applyPolicyOpt)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			applyPolicyRespBody, err := utils.FlattenResponse(applyPolicyResp)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+			jsonPath := fmt.Sprintf("central_network_policies[?id =='%s']|[0].is_applied", policyId)
+			statusRaw, err := jmespath.Search(jsonPath, applyPolicyRespBody)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error parse %s from response body", `central_network_policies.is_applied`)
+			}
+
+			status := fmt.Sprintf("%v", statusRaw)
+
+			targetStatus := []string{
+				"true",
+			}
+			if utils.StrSliceContains(targetStatus, status) {
+				return applyPolicyRespBody, "COMPLETED", nil
+			}
+
+			pendingStatus := []string{
+				"false",
+			}
+			if utils.StrSliceContains(pendingStatus, status) {
+				return applyPolicyRespBody, "PENDING", nil
+			}
+
+			return applyPolicyRespBody, status, nil
+		},
+		Timeout:      t,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func resourceCentralNetworkPolicyApplyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		getCentralNetworkPolicyApplyHttpUrl = "v3/{domain_id}/gcn/central-network/{central_network_id}/policies"
+		getCentralNetworkPolicyApplyProduct = "cc"
+	)
+	getCentralNetworkPolicyApplyClient, err := cfg.NewServiceClient(getCentralNetworkPolicyApplyProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CC client: %s", err)
+	}
+
+	getCentralNetworkPolicyApplyPath := getCentralNetworkPolicyApplyClient.Endpoint + getCentralNetworkPolicyApplyHttpUrl
+	getCentralNetworkPolicyApplyPath = strings.ReplaceAll(getCentralNetworkPolicyApplyPath, "{domain_id}", cfg.DomainID)
+	getCentralNetworkPolicyApplyPath = strings.ReplaceAll(getCentralNetworkPolicyApplyPath, "{central_network_id}", d.Id())
+
+	getCentralNetworkPolicyApplyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	getCentralNetworkPolicyApplyResp, err := getCentralNetworkPolicyApplyClient.Request("GET", getCentralNetworkPolicyApplyPath,
+		&getCentralNetworkPolicyApplyOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving central network policy")
+	}
+
+	getCentralNetworkPolicyApplyRespBody, err := utils.FlattenResponse(getCentralNetworkPolicyApplyResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getCentralNetworkPolicyApplyRespBody = utils.PathSearch("central_network_policies[?is_applied]|[0]",
```

Comment:
we should not retrieve the first policy if none policies are not applied?
```suggestion
	getCentralNetworkPolicyApplyRespBody = utils.PathSearch("central_network_policies[?is_applied]",
```

Reply:
It return nil if none policies are not applied, so it's ok